### DISCUSSION TITLE
GitOps media fact-checking team & GitOps community resources page

### DISCUSSION THREAD

**Discussion by scottrigby** [open-gitops/project#99] [Ideas]
## 📣 Update: New GitOps Community Resources Page! 

We now have a GitOps Community Resources Page in the project repo: https://github.com/open-gitops/project/blob/main/RESOURCES.md 🎉 

Everyone please take a look, and open PRs to add GitOps-related content! 👀✍️

Thanks to @samueltauil for putting the page together, it looks great 🤓💖

## To-do
- [x] Create Community Resources Page
- [ ] get updated commitments from folks who want to be on this team
- [ ] Create PR to move media fact-checking team to active

## Original issue

Hi everyone 👋 

## Motivation

Blog posts (and other resources like videos, talks, etc) on GitOps continue to be posted that do not align with the GitOps Principles officially published by the CNCF GitOps Working Group in October 2021 (see [v1.0.0](https://github.com/open-gitops/documents/releases/tag/v1.0.0)). A number of these resources seem to draw from older blog posts and resources that were around before the GitOps principles, and similarly, define GitOps totally different than the principles that now have nearly unilateral support across major cloud providers and other organizations.

## GitOps media fact-checking research team
Perhaps a GitOps fact-checking team could help this issue. The team's goal could be to promote accurate info about GitOps. This could be done through fact-checking, comparing writings and other resources to the principles, making these findings transparent, and contacting authors to invite them to make updates or addendums to their resources. I believe this aligns with the goals listed in the GitOps Working Group charter: https://github.com/cncf/tag-app-delivery/blob/main/gitops-wg/charter.md#goals

## GitOps community resources page
Perhaps we can also add a "GitOps community resources" page in one of the https://github.com/open-gitops repos, which lists blog posts and other resources that "need review" or "need revising or addendum" (since v1.0.0 of the GitOps principles was published). This would allow others to open PRs to list their published material too, and the PR review process could from that point forward be the process for checking the accuracy of information on GitOps in those resources.

WDYT?

### ⚠️ 💖 Proposed stance

It's important to note CNCF, it's TAGs or it's Working Groups do not have authority over ALL PEOPLE AND THINGS. So I suggest we take a warm, supportive – and not holier than thou – approach rather than anything that could feel like _"you're wrong and we somehow have the power to make you change your material"._ 

The WG and OpenGitOps project are meant to be open to anyone to participate on all levels, including reevaluating and proposing changes to the GitOps Principles as needed over time. I would like to see us invite folks to join us in this activity when it appears that they may have a different take, and to make them feel free to connect with us to revise posts they've already written. I hope people already feel free to connect with us during the process of creating or at least before publishing on GitOps if they wish, but maybe this team can help make this general invitation – to use the GitOps WG and OpenGitOps project as a resource – more widely known.

**Comment by mschwrdtnr** [comment]
Hey @scottrigby, I would be happy to participate for the `GitOps community resources page`. I am already researched for a larger number of GitOps resources like blogs and tutorials that follows the current GitOps principles and have a (semi-)structured list of it. If you have suggestions or ideas how we will manage a GitOps resource list and what the content or structure of this list should look like I can open a PR with my current resources :) 


## Some personal thoughts about a resource list
- According to what rules are resources included?
  - Have to follow all GitOps Principles? Have to follow at least three principles?
  - Include resources with proprietary software?
- Should resources separated by the focus of the resource, the used technologies, or something else?
- Should the list include all relevant resources or will only "the best" (in opinion of the community) included?
- How is it possible to keep the list up to date with the principles?

### Possible structure of a resource list
| Description | Focus of Resource | Technologies | GitOps Operator | Level / Comment | Link |
| --- | --- | --- | --- | --- | --- |
| Example GitOps Walkthrough with different repositories on Google Cloud Engine | Basic start tutorial | Docker, Kubernetes, GitHub, GitHub Actions, GKE, Terraform | Flux | Beginner | https://zwischenzugs.com/2021/07/31/a-hello-world-gitops-example-walkthrough/ |
| GitOps for multiple ML Environments @ SAP | Advanced technical knowledge sharing of production platform | Helm, Kustomize, Sealed Secrets, Argo Workflows | Argo CD | Advanced | https://blogs.sap.com/2021/05/06/decentralized-gitops-over-environments/ |
| Handle Secrets in GitOps with Bitnami Sealed Secrets | Secrets Management | Bitnami Sealed Secrets, Helm,...| - | Beginner | https://codefresh.io/about-gitops/handle-secrets-like-pro-using-gitops/ |

---
**I am not sure if these are necessary information and what information are also interesting like the type of the resource, date or something else. What do you think about my thoughts?**

**Reply by nishadmehendale** [reply]
Does this include courses available on learning platforms such as Pluralsight?

Slack Ref: https://cloud-native.slack.com/archives/C026D3RE7FB/p1650545053800629?thread_ts=1649953827.103189&cid=C026D3RE7FB

**Reply by mschwrdtnr** [reply]
Yeah, thats on of the questions we should answer in this discussion as mentioned in my thoughts.

**Reply by scottrigby** [reply]
Hi @mschwrdtnr 👋 appreciate the thoughtful reply 🙂

> Hey @scottrigby, I would be happy to participate for the `GitOps community resources page`. I am already researched for a larger number of GitOps resources like blogs and tutorials that follows the current GitOps principles and have a (semi-)structured list of it. If you have suggestions or ideas how we will manage a GitOps resource list and what the content or structure of this list should look like I can open a PR with my current resources :)

If it helps unblock anything I/we can make process suggestions, but ultimately IMO our processes should be nimble and easy to participate in have discussion around. In lieu of something else that feels better, we could (in any order):
- start collaborating on a simple doc, format unimportant (google doc, or if table feels better google sheet? No opinion on this, I just made a list because the extra columns get hidden in GItHub UI)
- could even try to collab through a GitHub beta project on this org (just use cards to start, turn into issues only if necessary. This could take advantage of things that are nicer than just a spreadsheet, and we could use templating too)
- check in with the wider GitOps Working Group during the twice monthly meetings, and Slack/email list in-between as needed. add an agenda item to the meeting doc for this so we don't forget. decide there what feels to be working and what does not, and to plan next steps

> ## Some personal thoughts about a resource list
> * According to what rules are resources included?
>   
>   * Have to follow all GitOps Principles? Have to follow at least three principles?
>   * Include resources with proprietary software?
> * Should resources separated by the focus of the resource, the used technologies, or something else?
> * Should the list include all relevant resources or will only "the best" (in opinion of the community) included?

re criteria: I think we could/should include any resource using the term "GitOps" in the title or content. Regardless of whether the term used purely for marketing purposes, or if others are genuinely attempting to put forth a different definition/standard, we should attempt to assume nothing but be as factual as possible. The current way I know to factually assess these is to see how the resource recommendations relate to the principles. For example, do they define GitOps as anything that uses Git in automation of any kind?

> * How is it possible to keep the list up to date with the principles?

I think any attempt would be good here to keep the ball rolling. Keep it simple so others can pick up where one person leaves off. We can also set an activity/absent policy like on other content for the group (if someone "claims" a task, but drops off for a week or something with no updates, someone else can pick back up on it). If we attempt to engage the authors, it would then be up to the authors whether they wish to update their materials in any way.

> 
> ### Possible structure of a resource list
> Description	Focus of Resource	Technologies	GitOps Operator	Level / Comment	Link
> Example GitOps Walkthrough with different repositories on Google Cloud Engine	Basic start tutorial	Docker, Kubernetes, GitHub, GitHub Actions, GKE, Terraform	Flux	Beginner	https://zwischenzugs.com/2021/07/31/a-hello-world-gitops-example-walkthrough/
> GitOps for multiple ML Environments @ SAP	Advanced technical knowledge sharing of production platform	Helm, Kustomize, Sealed Secrets, Argo Workflows	Argo CD	Advanced	https://blogs.sap.com/2021/05/06/decentralized-gitops-over-environments/
> Handle Secrets in GitOps with Bitnami Sealed Secrets	Secrets Management	Bitnami Sealed Secrets, Helm,...	-	Beginner	https://codefresh.io/about-gitops/handle-secrets-like-pro-using-gitops/
>
> **I am not sure if these are necessary information and what information are also interesting like the type of the resource, date or something else. What do you think about my thoughts?**

I like it. Let's just make sure we stay utilitarian, and set up as little busy-work as possible for folks to collaborate and achieve the goal. GitOps is a hot topic, so we'll need more hands to meet the demand. So maybe a rule of thumb is just add whatever feels useful at a glance without becoming overwhelming? Building on what you suggested, how about something like this (ordered by date most to least recent):

Resources:
- title/link: [Introduction to GitOps (LFS169)](https://training.linuxfoundation.org/training/introduction-to-gitops-lfs169/)
    - type: course
    - focus: tbd
    - published: date?
    - updated: date?
    - authors names and companies/orgs: Authors? (Linux Foundation)
    - technologies: tbd
    - skill level: beginner
    - audience: software developers 
    - assigned: @scottrigby 
    - status: to-do
    - notes: Scott: I saw this and another GitOps course by the LF, but unclear who put it together, when, or to what degree it conforms to the CNCF GitOps WG principles
-  [A ‘Hello World’ GitOps Example Walkthrough](https://zwischenzugs.com/2021/07/31/a-hello-world-gitops-example-walkthrough/)
    - type: blog post
    - focus: Basic start tutorial
    - published: 2021-07-31
    - updated: tbd
    - authors names and companies/orgs: [@ianmiell](https://twitter.com/ianmiell) (Container Solutions), [@someone_else](#) (Some Org)
    - technologies: Helm, Kustomize, Sealed Secrets, Argo Workflows
    - skill level: beginner
    - assigned @mschwrdtnr
    - status: in-progress
    - notes: initial list. Have not assessed or reached out

**Comment by todaywasawesome** [comment]
Love it, as long a we can do it with the right attitude and build bridges not create walls I think it's great.

**Comment by scottrigby** [comment]
Maybe better name – "GitOps media fact-checking team"?

**Reply by christianh814** [reply]
> Maybe better name – "GitOps media fact-checking team"?

I think that this could be part of a weekly "newsletter"? This media team can write a sort of "newsletter" that has a "fact checking" section.

Like 'this week in gitops'

**Reply by scottrigby** [reply]
Updated discussion title and description. This team's work is already in process. If anyone would like to join this team, reach out to us 🕵️ https://github.com/cncf/tag-app-delivery/tree/main/gitops-wg#community

**Comment by makkes** [comment]
I'd love to help out with the resources page or "this week in GitOps" (TWiG?). I got reminded of this discussion reading @rothgar's [awesome blog post](https://www.justingarrison.com/blog/2022-06-01-infrastructure-as-software?latest=true) on infrastructure as software and I'd love for this group to help spread the word on such great content.

**Comment by scottrigby** [comment]
## 📣 Update: New GitOps Community Resources Page! 

We now have a GitOps Community Resources Page in the project repo: https://github.com/open-gitops/project/blob/main/RESOURCES.md 🎉 

Everyone please take a look, and open PRs to add GitOps-related content! 👀✍️

Thanks to @samueltauil for putting the page together, it looks great 🤓💖

(also added to description at the top)
